docs/developer-guide/plugins-howto.md

```diff
@@ -330,6 +647,7 @@ Notice that also container related properties can be overridden in the applicati
  * Selectors
     * use existing selectors when possible to connect the state, eventually using reselect to compose them together or with your own selectors
     * define new selectors only if you use them more than once and move them to a selectors JS file
```

Comment:
I do not agree, putting anything in separate files decreases readability of the code, so we should put them in a separate file only when they are really shared.
Code re-use literally means "more than one use".

Reply:
I understand your point. please take into account also the following considerations:

1. When you develop a plugin, to understand if your selector already exists you have to look at all the plugins you think can have a similar behavior instead of look at the selector's file of the state splice you want to read. This of course encourages duplication. Instead having a file of already done function encourages the reuse of them.

2. Having the logic to access to the global state parts in the plugins decrease readability:

See this (not real) example of a tool that for epsg4326, if feature info is disabled, do geocoding on click : 

This is what you propose
```javascript
connect( () => createSelector(
  state => state.map && state.map.projection && state.map.prolection === "EPSG:4326",
  state => state.controls && state.controls.featureinfo && !state.controls.featureinfo.enabled ,
  state => state.geocoder && state.geocoder.services && state.geocoder.services.length > 0,
  (isCompatibleProjection, isFeatureInfoDisabled) => ({
  myProp: isCompatibleProjection && isFeatureInfoDisabled
}))
```

I think this version is more readable, hiding the state access complexity

```javascript
connect( () => createSelector(
  isGeocodingToolEnabled,
  enabled => ({enabled}) // map to your props
 )
```
this in selectors/geocoder.js: 
```javascript
/**
 * Check if the geocoder tool should be enabled (it is enabled only for  EPSG:4326, when at least 
 * one service is available and feature info has not been activated
 */
const isEnabled = createSelector(
  state => state.map && state.map.projection && state.map.projection === "EPSG:4326",
  state => state.controls && state.controls.featureinfo && !state.controls.featureinfo.enabled,
  state => state.geocoder && state.geocoder.services && state.geocoder.services.length > 0,
  (isCompatibleProjection, isFeatureInfoDisabled, hasServices) => isCompatibleProjection && FeatureInfoDisabled && hasServices
);
```
This: 
- easy to document and test
- ready for reuse
- easy to maintain
- easy to refactor
- With all the selectors in the same file you can quickly identify duplications and parts that can be reused inside the same selector. The above selector can be easly converted into

```javascript
const isEnabled = createSelector(
  state => projectionSelector(state) === "EPSG:4326", // from selectors/map.js
  state => !isControlEnabled("featureinfo")(state), // from controls
  state => servicesSelector(state).length > 0 // inside this file,
  (rightProj, isFeatureInfoDisabled, hasServices) => isCompatibleProjection && FeatureInfoDisabled && hasServices
);
```

Reply:
Does it means I convinced you or you're going to shoot me a rocket :laughing: ?